.github/workflows/publish.yml

```diff
@@ -0,0 +1,30 @@
+name: Upload Python Package to PyPI when a Release is Created
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  pypi-publish:
+    name: Publish release to PyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/pysequila
+    permissions:
+      id-token: write
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

Comment:
1. I can' see any automatic tagging and semantic versioning steps? 
2. It would be also nice to have a PR-workflow that run some basic checks like black, isort, and unit tests
3. pre-commit hook setup
4. It's not setup to run on our self-hosted runners
5. Would consider to use poetry instead (we try to use it in other projects as well) for the sake of lock file, cleaner way of managing dependencies and unification of development.

Reply:
I like to divide work to small deliverable parts. I wanted to first setup a basic publishing scheme, make sure the setup on the PyPI side is correct, and then add modifications one by one.

Reply:
Ok so please create follow-up issues that we won't forget about these points.